zebra/if_netlink.c

```diff
@@ -412,14 +413,19 @@ static uint32_t get_iflink_speed(struct interface *interface, int *error)
 				ifname, errno, safe_strerror(errno));
 		/* no device means interface unreachable */
 		if (errno == ENODEV && error)
-			*error = -1;
-		ecmd.speed_hi = 0;
-		ecmd.speed = 0;
+			*error = INTERFACE_SPEED_ERROR_READ;
+		return DEFAULT_BANDWIDTH;
 	}
 
 	close(sd);
 
-	return ((uint32_t)ecmd.speed_hi << 16) | ecmd.speed;
+	ret = ((uint32_t)(ecmd.speed_hi) << 16) | ecmd.speed;
```

Comment:
This is okay now, but the parentheses around `(ecmd.speed_hi)` are still unnecessary

Reply:
ok